libraries/docker_image.rb

```diff
@@ -16,12 +16,37 @@ class DockerImage < DockerBase
     property :rm, [true, false], default: true
     property :source, String
     property :tag, String, default: 'latest'
+    property :buildargs, [String, Array, Hash], coerce: proc { |v| coerce_buildargs(v) }
 
     alias_method :image, :repo
     alias_method :image_name, :repo
     alias_method :no_cache, :nocache
     alias_method :no_prune, :noprune
 
+    ###################
+    # Property helpers
+    ###################
+
+    def coerce_buildargs(v)
+      case v
+      when Hash
+        str = ''
+        v.each do |key, value|
+          str += '"' + key.to_s + '": "' + value.to_s + '",'
+        end
+        '{ ' + str.delete_suffix(',') + ' }'
+      when Array
+        str = ''
+        Array(v).each do |label|
+          parts = label.split(':')
+          str += '"' + parts[0].to_s + '": "' + parts[1].to_s + '",'
```

Comment:
This (and the Hash type) could be refactored to a map with a join at the end which wouldn't require stripping the extraneous comma at the end of the string as well. Using string interpolation rather than concat would be more readable too.
```suggestion
          v.map { |label| "\"#{label.split(':').first}\": \"#{label.split(':').last}\""}.join(',')
```